guides/common/modules/proc_updating-the-details-of-multiple-operating-systems.adoc

```diff
@@ -2,25 +2,25 @@
 = Updating the details of multiple operating systems
 
 Use this procedure to update the details of multiple operating systems.
-This example shows you how to assign each operating system a partition table called `Kickstart default`, a configuration template called `Kickstart default PXELinux`, and a provisioning template called `Kickstart Default`.
+This example shows you how to assign each operating system a partition table called `{client-provisioning-template-type} default`, a configuration template called `{client-provisioning-template-type} default PXELinux`, and a provisioning template called `{client-provisioning-template-type} Default`.
 
 .Procedure
 
 . On {ProjectServer}, run the following Bash script:
 +
-[source,terminal,options="nowrap" subs="+quotes"]
+[source, bash, options="nowrap" subs="+quotes,verbatim,attributes"]
 ----
-PARTID=$(hammer --csv partition-table list | grep "Kickstart default," | cut -d, -f1)
-PXEID=$(hammer --csv template list --per-page=1000 | grep "Kickstart default PXELinux" | cut -d, -f1)
-SATID=$(hammer --csv template list --per-page=1000 | grep "provision" | grep ",Kickstart default" | cut -d, -f1)
+PARTID=$(hammer --csv partition-table list | grep "{client-provisioning-template-type} default," | cut -d, -f1)
+PXEID=$(hammer --csv template list --per-page=1000 | grep "{client-provisioning-template-type} default PXELinux" | cut -d, -f1)
+{project-allcaps}_ID=$(hammer --csv template list --per-page=1000 | grep "provision" | grep ",{client-provisioning-template-type} default" | cut -d, -f1)
```

Comment:
Input on improving the templates is very much appreciated. For this PR, it is out of scope but I will keep this thread open and apply feedback later if necessary.

I have two possible follow-up changes in mind:

* Extract the actual code into snippets with a specific naming scheme, for example, `snip_code_x-y.adoc` and write some minor automation to at least lint the code using flake8/black/rubocop. Next step would be to automatically run this code on a Foreman/Katello instance.
* Deliver this code, similar to `bin/katello-certs-check` in foreman-installer, via RPM. We actually spoke about automating a procedure, I believe it was Salt or Ubuntu-related, in the past @ekohl.

Reply:
I'd prefer to recommend foreman-ansible-modules (or Red Hat's commercially supported branded equivalent). If you look at https://github.com/theforeman/forklift/pull/1856 you can see it's way easier to maintain.

Reply:
Yes, FAM works great. We use that downstream/internally quite a bit. However, as part of the _Foreman API_ guide, I don't think we want to show FAM examples (even though FAM also uses the Foreman/Katello API internally). I think a FAM example would be nice in https://docs.theforeman.org/nightly/Administering_Project/index-katello.html#Managing_Project_with_Ansible_Collections_admin

Is it OK to open an issue and move this discussion/implementation to another issue/PR?

Reply:
Hammer and FAM both use the API. If you're take that position, I think Hammer shouldn't be in the API guide either. But it's not: https://docs.theforeman.org/nightly/Provisioning_Hosts/index-katello.html#updating-the-details-of-multiple-operating-systems_provisioning

I think guiding users to FAM instead of writing shell scripts is a better solution, but agree it's out of scope for this PR.